ci/build.bat

```diff
@@ -6,19 +6,19 @@ if "%CONFIGURATION%" == "MINGW_32" (
   set ARCH=x86_64
   set BITS=64
   if "%CONFIGURATION%" == "MINGW_64-gcov" (
-    set USE_GCOV="-DUSE_GCOV=ON"
+    set "USE_GCOV=-DUSE_GCOV=ON"
   )
 ) else if "%CONFIGURATION%" == "MSVC_32" (
-  set CMAKE_GENERATOR="Visual Studio 15 2017"
+  set "CMAKE_GENERATOR=Visual Studio 15 2017"
 ) else if "%CONFIGURATION%" == "MSVC_64" (
-  set CMAKE_GENERATOR="Visual Studio 15 2017 Win64"
+  set "CMAKE_GENERATOR=Visual Studio 15 2017 Win64"
 )
 
 if "%CONFIGURATION:~0,5%" == "MINGW" (
   :: These are native MinGW builds, but they use the toolchain inside
   :: MSYS2, this allows using all the dependencies and tools available
   :: in MSYS2, but we cannot build inside the MSYS2 shell.
-  set CMAKE_GENERATOR="MinGW Makefiles"
+  set "CMAKE_GENERATOR=MinGW Makefiles"
```

Comment:
The quotes need to be part of the variable's value. They can be escaped like this:
```batch
set "CMAKE_GENERATOR=^"MinGW Makefiles^""
```
Or you could put quotes around the variable reference like this:
```batch
cmake -G "%CMAKE_GENERATOR%"
```

Would you accept a PR replacing this Batch script with a PowerShell script? I think it would be much easier to work with.

Reply:
> Would you accept a PR replacing this Batch script with a PowerShell script?

Definitely!

Reply:
The carets are unnecessary because it's the last argument. When using carets, it's best to use it for each metacharacter, including the double quotes used for wrapping the argument, in the entire command.

`set "CMAKE_GENERATOR="MinGW Makefiles""` is fine.